addons/test_website/tests/test_fuzzy.py

```diff
@@ -96,3 +98,34 @@ def test_02_pages_search(self):
                     'displayImage': False, 'allowFuzzy': True
                 }
             )
+    @patch('odoo.http.GeoIP')
+    def test_03_crm_partner_assign_geolocalization(self, GeoIpMock):
+        """
+            This test checks situation when "{OdooURL}/partners" is visited from foreign country without resellers.
+            It uses Mexico as an example.
+
+            Why patching of GeoIP is used?
+            Tested function (WebsiteCrmPartnerAssign.partners) uses GeoIp.country_code which is read_only, because
+            of property decorator https://docs.python.org/3/library/functions.html#property
+            Patching is allowing to modify normally read_only value.
+        """
+        # Patch GeoIp so it acts, as if Odoo client is located in Mexico
+        GeoIpMock.return_value.country_code = 'MX'
+
+        # Create a partner outside of Mexico
+        non_mexican_partner = self.env['res.partner'].create({
+            'name': 'Non_Mexican_Partner',
+            'is_company': True,
+            'grade_id': self.env['res.partner.grade'].search([], limit=1).id,
```

Comment:
#this model is added by website_crm_partner_assign so you can't use it in test_website (you can only use it in website_crm_partner_assign module or module that depends on website_crm_partner_assign)

same thing for `WebsiteCrmPartnerAssign` it's only usable those modules too

Reply:
good point! how about moving the test to `website_crm_partner_assign` module ?